src/sim-host/protocol/socket.js

```diff
@@ -11,52 +11,68 @@ module.exports.initialize = function (pluginHandlers, serviceToPluginMap) {
-
-        var service = data.service;
-        if (!service) {
-            throw 'Exec called on simulation host without a service specified';
-        }
-
-        var action = data.action;
-        if (!action) {
-            throw 'Exec called on simulation host without an action specified';
-        }
-
-        console.log('Exec ' + service + '.' + action + ' (index: ' + index + ')');
-
-        var handler = pluginHandlers[service] && pluginHandlers[service][action];
-        var telemetryProps = { plugin: serviceToPluginMap && serviceToPluginMap[service], service: service, action: action };
-        if (!handler) {
-            telemetryProps.handled = 'none';
-            handler = pluginHandlers['*']['*'];
-            handler(success, failure, service, action, data.args);
-        } else {
-            telemetryProps.handled = 'sim-host';
-            handler(success, failure, data.args);
-        }
-
-        telemetry.sendClientTelemetry('exec', telemetryProps);
-    });
+
     socket.on('refresh', function () {
         document.location.reload(true);
     });
+
+    socket.on('app-plugin-list', function (data) {
+        // TODO: process the list of plugins
+        socket.emit('start');
+    });
+
+    socket.once('init', function () {
+        socket.on('exec', function (data) {
+            var index;
+
+            if (!data) {
+                throw 'Exec called on simulation host without exec info';
+            }
+
+            index = data.index;
+            if (typeof index !== 'number') {
+                throw 'Exec called on simulation host without an index specified';
+            }
+
+
+            var success = data.hasSuccess ? getSuccess(index) : null;
+            var failure = data.hasFail ? getFailure(index) : null;
+            var service = data.service;
+            if (!service) {
+                throw 'Exec called on simulation host without a service specified';
+            }
+
+            var action = data.action;
+            if (!action) {
+                throw 'Exec called on simulation host without an action specified';
+            }
+
+            console.log('Exec ' + service + '.' + action + ' (index: ' + index + ')');
+
+            var handler = pluginHandlers[service] && pluginHandlers[service][action];
+            var telemetryProps = { plugin: serviceToPluginMap && serviceToPluginMap[service], service: service, action: action };
+            if (!handler) {
+                socket.emit('telemetry', {event: 'exec', props: {handled: 'none', service: service, action: action}});
+                handler = pluginHandlers['*']['*'];
+                handler(success, failure, service, action, data.args);
+            } else {
+                socket.emit('telemetry', {event: 'exec', props: {handled: 'sim-host', service: service, action: action}});
+                handler(success, failure, data.args);
+            }
+
+            telemetry.sendClientTelemetry('exec', telemetryProps);
```

Comment:
The current implementation of the telemetry-helper under should already take care of that. It will never send telemetry unless it has been told to initialize itself. So, as long as we only call `telemetryHelper.init()` when the server asks it, these calls will not be sent to the server needlessly.